runtime/src/main/java/org/corfudb/runtime/collections/TxnContext.java

```diff
@@ -95,18 +99,28 @@ private void txBeginInternal(boolean allowNestedTransactions) {
             log.warn("Reusing the transactional context created outside this layer!");
             this.iDidNotStartCorfuTxn = true;
             TransactionalContext.getRootContext().setTxnContext(this);
-            return;
+            return TransactionalContext.getRootContext().getSnapshotTimestamp();
         }
 
         log.trace("TxnContext: begin transaction in namespace {}", namespace);
         Transaction.TransactionBuilder transactionBuilder = this.objectsView
                 .TXBuild()
                 .type(TransactionType.WRITE_AFTER_WRITE);
+        Token snapshotToken;
         if (isolationLevel.getTimestamp() != Token.UNINITIALIZED) {
```

Comment:
quick question: Any specific reason for passing isolationLevel as a parameter rather than class level member variable?